homeassistant/components/opentherm_gw.py

```diff
@@ -0,0 +1,76 @@
+"""
+Support for OpenTherm Gateway devices.
+
+For more details about this component, please refer to the documentation at
+http://home-assistant.io/components/opentherm_gw/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.const import (CONF_DEVICE, CONF_NAME, PRECISION_HALVES,
+                                 PRECISION_TENTHS, PRECISION_WHOLE)
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+import homeassistant.helpers.config_validation as cv
+
+DOMAIN = 'opentherm_gw'
+
+CONF_CLIMATE = 'climate'
+CONF_FLOOR_TEMP = 'floor_temperature'
+CONF_PRECISION = 'precision'
+
+DATA_DEVICE = 'device'
+DATA_GW_VARS = 'gw_vars'
+DATA_OPENTHERM_GW = 'opentherm_gw'
+
+SIGNAL_OPENTHERM_GW_UPDATE = 'opentherm_gw_update'
+
+CLIMATE_SCHEMA = vol.Schema({
+    vol.Optional(CONF_NAME, default="OpenTherm Gateway"): cv.string,
+    vol.Optional(CONF_PRECISION): vol.In([PRECISION_TENTHS, PRECISION_HALVES,
+                                          PRECISION_WHOLE]),
+    vol.Optional(CONF_FLOOR_TEMP, default=False): cv.boolean,
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_DEVICE): cv.string,
+        vol.Optional(CONF_CLIMATE, default=CLIMATE_SCHEMA({})):
```

Comment:
Just pass an empty dict as default value. It will be automatically passed to the schema.